src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1826,10 +1867,19 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode,
 
 		if (hasAggregates)
 		{
+			walkerContext->pullDistinctColumns = ShouldPullDistinctColumn(
+				groupedByDisjointPartitionColumn, repartitionSubquery,
+				hasNonPartitionColumnDistinctAgg,
+				(Node *) originalExpression, tableNodeList);
+
 			WorkerAggregateWalker((Node *) originalExpression, walkerContext);
 
 			newExpressionList = walkerContext->expressionList;
 			queryHasAggregates = true;
+			if (walkerContext->createGroupByClause)
```

Comment:
explaining why this is the case with a comment and putting a new line before the `if` would be nice

Reply:
added comments, moved to the group creation block below